crates/wysiwyg/src/composer_model/format.rs

```diff
@@ -74,18 +74,14 @@ where
     }
 
     pub(crate) fn apply_pending_formats(&mut self, start: usize, end: usize) {
-        self.state
-            .toggled_format_types
-            .clone()
-            .iter()
-            .for_each(|format| {
-                if self.reversed_actions.contains(&format.action()) {
-                    self.format_range(start, end, format);
-                } else {
-                    self.unformat_range(start, end, format);
-                }
-            });
-        self.state.toggled_format_types.clear();
+        while self.state.toggled_format_types.len() > 0 {
```

Comment:
you should be able to replace the` while` and the line immediately after with:

```rust
for format in self.state.toggled_format_types.into_iter() {
```

Reply:
This adds some borrowing issue for me, and I don't see how it will remove from the array at the same time ? 
I changed a bit, with a reverse + pop instead of removing 0 potentially multiple times.